raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -495,6 +496,31 @@ public void testHandleBeginQuorumEpochAfterUserInitiatedResign() throws Exceptio
             context.listener.currentLeaderAndEpoch());
     }
 
+    @Test
+    public void testBeginQuorumHeartbeat() throws Exception {
+        int localId = 0;
+        int remoteId1 = 1;
+        int remoteId2 = 2;
+        Set<Integer> voters = Utils.mkSet(localId, remoteId1, remoteId2);
+
+        RaftClientTestContext context = new RaftClientTestContext.Builder(localId, voters).build();
+
+        context.becomeLeader();
+        assertEquals(OptionalInt.of(localId), context.currentLeader());
+
+        // begin epoch requests should be sent out every beginQuorumEpochTimeoutMs
+        context.time.sleep(context.beginQuorumEpochTimeoutMs);
+        context.client.poll();
+        context.expectBeginEpoch(context.currentEpoch());
```

Comment:
It looks `expectBeginEpoch` already `pollUntilRequests`. Having said that `expectBeginEpoch` is not precise enough. We need to check that the leader sent a BEGIN_QUORUM_EPOC request to all of the voters.